lib/mayaUsd/ufe/UsdStageMap.h

```diff
@@ -27,33 +27,31 @@
 
 #include <unordered_map>
 
-// Allow for use of MObjectHandle with std::unordered_map.
-namespace std {
-template <> struct hash<MObjectHandle>
-{
-    std::size_t operator()(const MObjectHandle& handle) const
-    {
-        return static_cast<std::size_t>(handle.hashCode());
-    }
-};
-} // namespace std
+// Pending rework of mayaUsd namespaces, MayaUsdProxyShapeBase is in the Pixar
+// namespace.  PPT, 9-Mar-2021.
+PXR_NAMESPACE_OPEN_SCOPE
+class MayaUsdProxyShapeBase;
+PXR_NAMESPACE_CLOSE_SCOPE
 
 namespace MAYAUSD_NS_DEF {
 namespace ufe {
 
 //! \brief USD Stage Map
 /*!
-        Two-way map of proxy shape UFE path to corresponding stage.
-
-        We will assume that	a USD proxy shape will not be instanced (even though
-        nothing in the data model prevents it).  To support renaming and repathing,
-        we store an MObjectHandle in the maps, which is invariant to renaming and
-        repathing, and compute the path on access.  This is slower than a scheme
-        where we cache using the Ufe::Path, but such a cache must be refreshed on
-        rename and repath, which is non-trivial, since there is no guarantee on the
-        order of notification of Ufe observers.  An earlier implementation with
-        rename observation had the Maya Outliner (which observes rename) access the
-        UsdStageMap on rename before the UsdStageMap had been updated.
+    Two-way map of proxy shape UFE path to corresponding stage.
+
+    We will assume that	a USD proxy shape will not be instanced (even though
+    nothing in the data model prevents it).  To generalized access to the
+    underlying node, we store an MObjectHandle in the maps.
+
+    The cache is refreshed on access to a stage given a path which cannot be
```

Comment:
Still wish to avoid having the cache observe the data model, to avoid observation cost as well as order of notification problems.  Rebuild on access failure is simpler to code, and access failure through incorrect input is most likely not a high-frequency code path.